src/libraries/System.Threading.RateLimiting/tests/Infrastructure/Utils.cs

```diff
@@ -13,7 +13,7 @@ namespace System.Threading.RateLimiting.Tests
 {
     internal static class Utils
     {
-        internal static Func<Task> StopTimerAndGetTimerFunc<T>(PartitionedRateLimiter<T> limiter)
+        internal static async Task<Func<Task>> StopTimerAndGetTimerFunc<T>(PartitionedRateLimiter<T> limiter)
```

Comment:
This type signature! 😆 It looks like this should fix the bug, but can we just add a private/internal ctor to `DefaultPartitionedRateLimiter` that doesn't start the timer?

Reply:
Tried out the private ctor approach. Definitely removes the possibility of test races now.